### PR TITLE
Bump cleanup signal handling before anything else

### PR DESCRIPTION
**Pull request by rdimitrov**
Signed-off-by: Radoslav Dimitrov <dimitrovr@vmware.com>

## How Has This Been Tested?

- [ ] Covered by existing integration testing
- [ ] Added integration testing to cover
- [ ] Tested locally
- [ ] Have not tested

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

**Comment by lobkovilya** [comment]
Probably order of health checks should be changed also in https://github.com/networkservicemesh/networkservicemesh/blob/master/controlplane/cmd/nsmd/nsmd.go, because `logrus.Fatalf` calls `os.Exit(1)`

**Comment by rdimitrov** [comment]
@lobkovilya Thanks for the feedback 👍 
Do you suggest swapping the `Fatalf` with `Warnf` or to put the health check after everything fatal-related?
Basically, even now in case the `Fatalf` exits, the probing server will die causing the probes checking to fail as well.

**Comment by lobkovilya** [comment]
As for me, I believe that behavior of `Fatalf` is not obvious enough, so I'd prefer `Warnf` or `Errorf` and simple `return`.
Also to avoid duplicates I think it's better to call `defer nsmd.SetAPIServerFailed()` at the beginning

File: dataplane/vppagent/cmd/vppagent-dataplane.go [hunk 1]
```diff
@@ -35,6 +33,8 @@ func main() {
 		syscall.SIGTERM,
 		syscall.SIGQUIT)
 
+	go common.BeginHealthCheck()
```
Comment (edwarnicke) [inline dataplane/vppagent/cmd/vppagent-dataplane.go]: Thanks for fixing this... I'm sensitive to it because I've been bitten by it before ;)

**Pull request merged**